src/bootstrap/test.rs

```diff
@@ -966,6 +966,15 @@ impl Step for Compiletest {
     /// compiletest `mode` and `suite` arguments. For example `mode` can be
     /// "run-pass" or `suite` can be something like `debuginfo`.
     fn run(self, builder: &Builder<'_>) {
+        if builder.top_stage == 0 && env::var("COMPILETEST_FORCE_STAGE0").is_err() {
+            eprintln!("\
+error: `--stage 0` is not meaningful for compiletest test suites
```

Comment:
I don't think we should say "not meaningful" - it *is* meaningful, it's just almost never going to work, because beta and master have almost always already diverged sufficiently to break tests.

Reply:
How does this sound?

> error: `--stage 0` runs compiletest on the beta compiler, not your local changes, and will almost always cause tests to fail